infrastructure/terraform/modules/templates-api/spec.tmpl.json

```diff
@@ -58,9 +273,169 @@
           "authorizerUri": "arn:aws:apigateway:${AWS_REGION}:lambda:path/2015-03-31/functions/${AUTHORIZER_LAMBDA_ARN}/invocations",
-          "authorizerCredentials": "${APIG_EXECUTION_ROLE_ARN}"
+          "authorizerCredentials": "${APIG_EXECUTION_ROLE_ARN}",
+          "identitySource": "method.request.header.authorization,context.resourceId,context.httpMethod",
+          "authorizerResultTtlInSeconds": 300
+        }
+      }
+    },
+    "schemas": {
+      "TemplateDTO": {
+        "type": "object",
+        "required": [
+          "id",
+          "templateType",
+          "templateStatus",
+          "name",
+          "message",
+          "createdAt",
+          "updatedAt"
+        ],
+        "properties": {
+          "id": {
+            "type": "string"
+          },
+          "templateType": {
+            "$ref": "#/components/schemas/TemplateType"
+          },
+          "templateStatus": {
+            "$ref": "#/components/schemas/TemplateStatus"
+          },
+          "name": {
+            "type": "string"
+          },
+          "message": {
+            "type": "string"
+          },
+          "createdAt": {
+            "type": "string",
+            "format": "date-time"
+          },
+          "updatedAt": {
+            "type": "string",
+            "format": "date-time"
+          },
+          "subject": {
+            "type": "string"
+          }
+        }
+      },
+      "TemplateType": {
+        "type": "string",
+        "enum": [
+          "NHS_APP",
+          "SMS",
+          "EMAIL"
+        ]
+      },
+      "TemplateStatus": {
+        "type": "string",
+        "enum": [
+          "NOT_YET_SUBMITTED",
+          "SUBMITTED"
+        ]
+      },
+      "CreateTemplate": {
+        "type": "object",
+        "required": [
+          "templateType",
+          "name",
+          "message"
+        ],
+        "properties": {
+          "templateType": {
+            "$ref": "#/components/schemas/TemplateType"
+          },
+          "name": {
+            "type": "string"
+          },
+          "message": {
+            "type": "string"
+          },
+          "subject": {
+            "type": "string"
+          }
+        }
+      },
+      "UpdateTemplate": {
+        "type": "object",
+        "required": [
+          "templateStatus",
+          "name",
+          "message",
+          "templateType"
+        ],
+        "properties": {
+          "templateStatus": {
+            "$ref": "#/components/schemas/TemplateStatus"
+          },
+          "name": {
+            "type": "string"
+          },
+          "message": {
+            "type": "string"
+          },
+          "subject": {
+            "type": "string"
+          },
+          "templateType": {
+            "$ref": "#/components/schemas/TemplateType",
+            "readOnly": true,
+            "description": "This value will never be updated. It is used to determine the type of template being validated."
```

Comment:
if this is never updated we probably don't want it in the API request - the API can figure out for itself how to validate an update to an already existing template without the user telling them

Reply:
The purpose of it being here is use the correct zod schema when validating the template. 

If we remove this field then I'd need to fetch the item from the database before I do the update or split the API into /templates/email,sms,app. 

My thinking was this was a seemingly decent middle ground. What do you think?

Reply:
We'll be fetching the data anyway to do an authorisation check, i.e. are you allowed to update it?

Reply:
Chatted on teams about this.

Decided leaving this as-is is a good enough middle ground.